compiler/tests/org/jetbrains/kotlin/cli/FriendPathsTest.kt

```diff
@@ -38,6 +38,21 @@ class FriendPathsTest : TestCaseWithTmpdir() {
         )
     }
 
+    /** Regression test for KT-34251. */
+    fun testArchive_relativePath() {
+        val libSrc = File(getTestDataDirectory(), "lib.kt")
+        val libDest = File(tmpdir, "lib.jar").relativeTo(File("").absoluteFile)
+        CompilerTestUtil.executeCompilerAssertSuccessful(K2JVMCompiler(), listOf("-d", libDest.path, libSrc.path))
+
+        CompilerTestUtil.executeCompilerAssertSuccessful(
+            K2JVMCompiler(),
+            listOf(
+                "-d", tmpdir.path, "-cp", libDest.path, File(getTestDataDirectory(), "usage.kt").path,
+                "-Xfriend-paths=${libDest.path}"
+            )
+        )
+    }
```

Comment:
Instead of copy-pasting, could you please extract the common part with `testArchive` into a function (something like `fun doTestArchive(relative: Boolean)`), and similarly for `testDirectory` tests?